jms/src/main/scala/akka/stream/alpakka/jms/JmsBrowseStage.scala

```diff
@@ -10,9 +10,8 @@ import akka.stream.{ActorAttributes, Attributes, Outlet, SourceShape}
 import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
 import java.util.{Enumeration => JEnumeration}
 
-private[jms] final class JmsBrowseStage(settings: JmsBrowseSettings) extends GraphStage[SourceShape[Message]] {
-  private val queue = settings.destination.getOrElse { throw new IllegalArgumentException("Destination is missing") }
-
+private[jms] final class JmsBrowseStage(settings: JmsBrowseSettings, queue: Destination)
```

Comment:
Destination is now checked on flow creation, and passed explicitly into the stages